File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    bool is_number(const string& s) {
        return !s.empty() && (isdigit(s[0]) || (s[0] == '-' && s.size() > 1));
    }
    int evalRPN(vector<string>& tokens) {
        stack<int> st;
        for (int i = 0; i < tokens.size(); i++) {
            string ch = tokens[i];
            if (is_number(ch)) {
                st.push(stoi(ch));
            } else {
                int a = st.top();
                st.pop();
                int b = st.top();
                st.pop();
                if (ch == "+") {
                    int c = b + a;
                    st.push(c);
                }
                if (ch == "-") {
                    int c = b- a;
                    st.push(c);
                }
                if (ch == "*") {
                    int c = b*a;
                    st.push(c);
                }
                if (ch == "/") {
                    int c = b/a;
                    st.push(c);
                }
            }
        }
        return st.top();
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for token in tokens:
            x = token.strip("-")
            if x.isalnum():
                stack.append(int(token))
            else:
                x, y = stack.pop(), stack.pop()
                if token == "+":
                    stack.append(int(x+y))
                if token == "-":
                    stack.append(int(y-x))
                if token == "/":
                    stack.append(int(y/x))
                if token == "*":
                    stack.append(int(x*y))
        return stack[-1]
                
    
                
                    

